pandas/tests/indexes/datetimes/test_date_range.py

```diff
@@ -290,6 +290,24 @@ def test_precision_finer_than_offset(self):
         tm.assert_index_equal(result1, expected1)
         tm.assert_index_equal(result2, expected2)
 
+    def test_mismatching_tz_raises_err(self):
+        dt1_tz = pd.Timestamp('2017-01-01', tz='US/Eastern')
+        dt1_no_tz = pd.Timestamp('2017-01-01')
+        dt2_tz = pd.Timestamp('2017-01-04', tz='US/Eastern')
```

Comment:
can you parametrize this test (IOW pass in start, end), cases for raising include:

- one naive, one with a tz & reversed
- different timezones & reversed

I don't think we care about the freq here.